File: implement-queue-using-stacks/implement-queue-using-stacks.py
from collections import deque
class MyQueue:
    def __init__(self):
        self.dq = deque()
        

    def push(self, x: int) -> None:
        self.dq.append(x)

    def pop(self) -> int:
        return self.dq.popleft() if self.dq else None

    def peek(self) -> int:
        return self.dq[0]

    def empty(self) -> bool:
        return len(self.dq) == 0

File: implement-queue-using-stacks/implement-queue-using-stacks.cpp

class MyQueue {
public:
    stack<int> mainStack;
    stack<int> auxStack;

    MyQueue() {
    }
    
    void push(int x) {
        while (!mainStack.empty()) {
            int top = mainStack.top();
            mainStack.pop();
            auxStack.push(top);
        }
        mainStack.push(x);
        while(!auxStack.empty()) {
            int top = auxStack.top();
            auxStack.pop();
            mainStack.push(top);
        }
    }
    
    int pop() {
        int top = mainStack.top();
        mainStack.pop();
        return top;
    }
    
    int peek() {
        return mainStack.top();
    }
    
    bool empty() {
        return mainStack.empty();
    }
};
